core/libraries/plugin_api/EE_Register_Model.lib.php

```diff
@@ -129,52 +132,59 @@ public static function register($model_id = null, $config = array())
                 sprintf(__("The key '%s' is not a known key for registering a model", "event_espresso"), $unknown_key)
             );
         }
+        return true;
     }
 
+
     /**
      * Filters the core list of models
      *
      * @param array $core_models
      * @return array keys are model names (eg 'Event') and values are their classes (eg 'EE_Event')
      */
-    public static function add_addon_models($core_models = array())
+    public static function add_addon_models($core_models = []): array
     {
-        foreach (self::$_model_name_to_classname_map as $model_name_to_class_map) {
-            $core_models = array_merge($core_models, $model_name_to_class_map);
+        $models = [];
+        foreach (self::$_model_name_to_classname_map as $model_map) {
+            $models[] = $model_map;
         }
-        return $core_models;
+        return array_merge($core_models, ...$models);
     }
 
+
     /**
      * Filters the list of model folders
      *
      * @param array $folders
      * @return array of folder paths
      */
-    public static function add_model_folders($folders = array())
+    public static function add_model_folders(array $folders = []): array
     {
+        $model_folders = [];
         foreach (self::$_model_registry as $config) {
             if (isset($config['model_paths'])) {
-                $folders = array_merge($folders, $config['model_paths']);
+                $model_folders[] = $config['model_paths'];
             }
         }
-        return $folders;
+        return array_merge($folders, ...$model_folders);
     }
 
+
     /**
      * Filters the array of model class paths
      *
      * @param array $folders
      * @return array of folder paths
      */
-    public static function add_class_folders($folders = array())
+    public static function add_class_folders(array $folders = []): array
     {
+        $class_folders = [];
         foreach (self::$_model_registry as $config) {
             if (isset($config['class_paths'])) {
-                $folders = array_merge($folders, $config['class_paths']);
+                $class_folders[] = $config['class_paths'];
             }
         }
-        return $folders;
+        return array_merge($folders, ...$class_folders);
```

Comment:
Spread Operator in Arrays is available only in PHP >= 7.4. RFC [here](https://wiki.php.net/rfc/spread_operator_for_array).

Reply:
That's to allow the use of the spread operator during array definition,
but the spread operator has been available for use in argument unpacking since 5.6:
https://www.php.net/manual/en/migration56.new-features.php#migration56.new-features.splat